crates/biome_js_analyze/src/lint/nursery/no_unknown_attribute.rs

```diff
@@ -0,0 +1,1206 @@
+    "vHanging",
+    "vIdeographic",
+    "vMathematical",
+    "vectorEffect",
+    "vertAdvY",
+    "vertOriginX",
+    "vertOriginY",
+    "viewBox",
+    "viewTarget",
+    "wordSpacing",
+    "writingMode",
+    "xChannelSelector",
+    "xHeight",
+    "xlinkActuate",
+    "xlinkArcrole",
+    "xlinkHref",
+    "xlinkRole",
+    "xlinkShow",
+    "xlinkTitle",
+    "xlinkType",
+    "xmlBase",
+    "xmlLang",
+    "xmlnsXlink",
+    "xmlSpace",
+    "yChannelSelector",
+    "zoomAndPan",
+];
+
+fn is_valid_data_attribute(name: &str) -> bool {
+    use biome_string_case::StrOnlyExtension;
+    if !name.starts_with("data-") {
+        return false;
+    }
+
+    if name.to_lowercase_cow().starts_with("data-xml") {
+        return false;
+    }
+
+    let data_name = &name["data-".len()..];
+
+    if data_name.is_empty() {
+        return false;
+    }
+
+    data_name.chars().all(|c| c != ':')
+}
+
+fn is_valid_aria_attribute(name: &str) -> bool {
+    ARIA_PROPERTIES.iter().any(|&element| element == name)
+}
+
+fn is_valid_html_tag_in_jsx(node: &AnyJsxElement, tag_name: &str) -> bool {
+    let matches_tag_convention = tag_name.char_indices().all(|(i, c)| {
+        if i == 0 {
+            c.is_ascii_lowercase()
+        } else {
+            c != '-'
+        }
+    });
+
+    if matches_tag_convention {
+        return node.attributes().find_by_name("is").is_none();
+    }
+
+    false
+}
+
+fn tag_name_has_dot(node: &AnyJsxElement) -> Option<bool> {
+    Some(matches!(
+        node.name().ok()?,
+        AnyJsxElementName::JsxMemberName(_)
+    ))
+}
+
+fn has_uppercase(name: &str) -> bool {
+    name.contains(char::is_uppercase)
+}
+
+pub enum NoUnknownAttributeState {
+    UnknownProp {
+        name: Box<str>,
+    },
+    UnknownPropWithStandardName {
+        name: Box<str>,
+        standard_name: Box<str>,
+    },
+    InvalidPropOnTag {
+        name: Box<str>,
+        tag_name: TokenText,
+        allowed_tags: &'static [&'static str],
+    },
+    DataLowercaseRequired {
+        name: Box<str>,
+    },
+}
+
+fn get_standard_name(name: &str) -> Option<&'static str> {
+    if let Some(&standard_name) = DOM_ATTRIBUTE_LOOKUP.get(name) {
+        return Some(standard_name);
+    }
+
+    DOM_PROPERTY_NAMES
+        .iter()
+        .find(|&&element| element.eq_ignore_ascii_case(name))
+        .copied()
+}
+
+impl Rule for NoUnknownAttribute {
+    type Query = Manifest<JsxAttribute>;
+    type State = NoUnknownAttributeState;
+    type Signals = Option<Self::State>;
+    type Options = NoUnknownAttributeOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        let options = ctx.options();
+
+        let node_name = match node.name().ok()? {
+            AnyJsxAttributeName::JsxName(name) => name.syntax().text_trimmed(),
+            AnyJsxAttributeName::JsxNamespaceName(name) => name.syntax().text_trimmed(),
+        };
+        let node_name = node_name.to_string();
```

Comment:
Here's how to avoid the string allocation for namespaced attributes:

You can put them in a separate map like this, where the key is a tuple of `(&str, &str)`. The full type is `FxHashMap<(&str, &str), &str>`

```rust
let test = FxHashMap::from_iter([
            (("xlink", "actuate"), "xlinkActuate"),
            (("xlink", "arcrole"), "xlinkArcrole"),
            (("xlink", "href"), "xlinkHref"),
            (("xlink", "role"), "xlinkRole"),
            (("xlink", "show"), "xlinkShow"),
            (("xlink", "title"), "xlinkTitle"),
            (("xlink", "type"), "xlinkType"),
            (("xml", "base"), "xmlBase"),
            (("xml", "lang"), "xmlLang"),
            (("xml", "space"), "xmlSpace"),
        ]);
```

Reply:
@dyc3 i used `into_text()` method instead

```rust
        let node_name = match node.name().ok()? {
            AnyJsxAttributeName::JsxName(name) => name.syntax().text_trimmed(),
            AnyJsxAttributeName::JsxNamespaceName(name) => name.syntax().text_trimmed(),
        };

        let node_name = node_name.into_text();
```